rust/agents/scraper/src/scraper/mod.rs

```diff
@@ -795,39 +824,53 @@ impl SqlChainScraper {
             let _ = blocks
                 .get_mut(&hash)
                 .expect("We found a block that we did not request")
-                .insert((Some(block.id), block.timestamp));
+                .insert((
+                    Some(block.id),
+                    BlockInfo {
+                        hash,
+                        timestamp: date_time::to_unix_timestamp_s(&block.timestamp),
+                        // TODO: we don't actually use these below, we should make sure to clean
+                        // this out
+                        number: 0,
+                        gas_used: Default::default(),
+                        gas_limit: Default::default(),
+                    },
+                ));
         }
 
         let blocks_to_fetch = blocks
             .iter_mut()
             .inspect(|(_, info)| {
                 // info being defined implies the id has been set (at this point)
-                debug_assert!(info.is_none() || info.unwrap().0.is_some())
+                debug_assert!(info.is_none() || info.as_ref().unwrap().0.is_some())
             })
             .filter(|(_, block_info)| block_info.is_none());
-        for (_hash, block_info) in blocks_to_fetch {
-            // TODO: fetch block data from ethers
-            // (timestamps are not included in LogMeta)
-            let _ = block_info.insert((None, crate::date_time::now()));
+        for (hash, block_info) in blocks_to_fetch {
+            let info = self.local.provider.get_block_by_hash(hash).await?;
+            let _ = block_info.insert((None, info));
```

Comment:
still find the tuple usage hard to read-- iirc there are plans to refactor?

Reply:
Yes, this has been cleaned up significantly in my new branch. I will raise a PR once this branch is merged.